From 4fe71f74c6e8736a874cd8c9b207ac987904c46b Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 07 Jun 2022 19:28:49 +0300
Subject: [PATCH] [SW-0] gaudi3: add progress indications in rom recovery

This patch adds additional progress indications in rom recovery
routine.

Change-Id: I04a0abf330d3f66afdfaeb5377b0c2317c581d28
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index be55887..08215bf 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -170,6 +170,8 @@
 		rom_pcie_recovery();
 
 	rom_d2d_link_init();
+
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_RECOVERY_END);
 }
 
 static void execute_erom_recovery(void)
diff --git a/common/pcie.c b/common/pcie.c
index 83b0fc7..b5f5770 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -80,6 +80,8 @@
 	uint32_t link_ctrl;
 	uint8_t ltssm_en;
 
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_START);
+
 	pcie_loop_while_prstn(PCIE_PRSTN_ASSERT);
 
 	pcie_set_pcie_device_vendor_id();
@@ -88,6 +90,7 @@
 				LTSSM_EN_IND_MASK;
 
 	if (!ltssm_en) {
+		arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT);
 		regmap_writel(0, mmPCIE_PHY_RST_N);
 
 		pcie_phy_para_sel_set();
@@ -108,6 +111,7 @@
 		pcie_phy_set_sram_ext_ld_done();
 
 		if ((arcboot_rom_global.config_fuse & EFUSE_SYSCFG_PCIE_GEN_LIMIT) == 0) {
+			arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1);
 			/* limit PCIe link gen to 1 */
 			link_ctrl = regmap_readl(mmLINK_CAPABILITIES_REG);
 			link_ctrl &=
@@ -125,6 +129,7 @@
 
 		/* Configure IATU */
 		if (!arcboot_rom_global.die_id) {
+			arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG);
 			pcie_config_iatu();
 			pcie_config_boot_termination();
 		}
@@ -139,6 +144,8 @@
 
 		regmap_writel(0x3, mmLTSSM_EN);
 	}
+
+	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH);
 }
 
 void pcie_recovery(void)
diff --git a/include/arcboot.h b/include/arcboot.h
index 09de156..8e87eee 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -209,65 +209,71 @@
 	PROGRESS_ARCBOOT_RECOVERY = 67,
 	PROGRESS_ARCBOOT_RECOVERY_DMA = 68,
 	PROGRESS_ARCBOOT_IN_ROM_RECOVERY = 69,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 70,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 71,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 72,
-	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 73,
-	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 74,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_START = 75,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_DEST = 76,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 77,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_ERROR = 78,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_FINISH = 79,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 80,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_VERIFICATION_FAIL = 81,
-	PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR = 82,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 83,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 84,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 85,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 86,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 87,
-	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 88,
-	PROGRESS_ARCBOOT_EROM_PCIE = 89,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 90,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 91,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 92,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 93,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 94,
-	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 95,
-	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 96,
-	PROGRESS_ARCBOOT_EROM_PCIE_START = 97,
-	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 98,
-	PROGRESS_ARCBOOT_EROM_RECOVERY = 99,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 100,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 101,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 102,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 103,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 104,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 105,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 106,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 107,
-	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 108,
-	PROGRESS_ARCBOOT_EROM_TPM = 109,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 110,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 111,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 112,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 113,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 114,
-	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 115,
-	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 116,
-	PROGRESS_ARCBOOT_EROM_TPM_START = 117,
-	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 118,
-	PROGRESS_ARCBOOT_EROM_IMG0 = 119,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 120,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 121,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 122,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 123,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 124,
-	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 125,
-	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 126,
-	PROGRESS_ARCBOOT_EROM_IMG0_START = 127,
-	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 128,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_START = 70,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_INIT = 71,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_LIMIT_GEN1 = 72,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_IATU_CONFIG = 73,
+	PROGRESS_ARCBOOT_ROM_PCIE_LINK_FINISH = 74,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_START = 75,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_STS_BUSY = 76,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_TIMEDOUT = 77,
+	PROGRESS_ARCBOOT_ROM_D2D_LINK_INIT_SUCCESS = 78,
+	PROGRESS_ARCBOOT_ROM_RECOVERY_END = 79,
+	PROGRESS_ARCBOOT_IN_EROM_RECOVERY = 80,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_START = 81,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_DEST = 82,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_UNALIGNED_IMG_SIZE = 83,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_ERROR = 84,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_DMA_FINISH = 85,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_HDR_MAGIC_FAIL = 86,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_VERIFICATION_FAIL = 87,
+	PROGRESS_ARCBOOT_EROM_BOOT_DESC_ERROR = 88,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_START = 89,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_VERSION_FAIL = 90,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_REVOKED = 91,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_KEY_HASH_FAIL = 92,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_AUTH_FAIL = 93,
+	PROGRESS_EROM_BOOT_DESC_VERIFY_SUCCESS = 94,
+	PROGRESS_ARCBOOT_EROM_PCIE = 95,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_START = 96,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_CRC_FAILED = 97,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_DEST = 98,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_UNALIGNED_IMG_SIZE = 99,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_ERROR = 100,
+	PROGRESS_ARCBOOT_EROM_PCIE_DMA_FINISH = 101,
+	PROGRESS_ARCBOOT_EROM_PCIE_VERIFICATION_FAIL = 102,
+	PROGRESS_ARCBOOT_EROM_PCIE_START = 103,
+	PROGRESS_ARCBOOT_EROM_PCIE_FINISH = 104,
+	PROGRESS_ARCBOOT_EROM_RECOVERY = 105,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_START = 106,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_CRC_FAILED = 107,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_DEST = 108,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_UNALIGNED_IMG_SIZE = 109,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_ERROR = 110,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_DMA_FINISH = 111,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_VERIFICATION_FAIL = 112,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_START = 113,
+	PROGRESS_ARCBOOT_EROM_RECOVERY_FINISH = 114,
+	PROGRESS_ARCBOOT_EROM_TPM = 115,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_START = 116,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_CRC_FAILED = 117,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_DEST = 118,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_UNALIGNED_IMG_SIZE = 119,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_ERROR = 120,
+	PROGRESS_ARCBOOT_EROM_TPM_DMA_FINISH = 121,
+	PROGRESS_ARCBOOT_EROM_TPM_VERIFICATION_FAIL = 122,
+	PROGRESS_ARCBOOT_EROM_TPM_START = 123,
+	PROGRESS_ARCBOOT_EROM_TPM_FINISH = 124,
+	PROGRESS_ARCBOOT_EROM_IMG0 = 125,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_START = 126,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_CRC_FAILED = 127,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_DEST = 128,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_UNALIGNED_IMG_SIZE = 129,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_ERROR = 130,
+	PROGRESS_ARCBOOT_EROM_IMG0_DMA_FINISH = 131,
+	PROGRESS_ARCBOOT_EROM_IMG0_VERIFICATION_FAIL = 132,
+	PROGRESS_ARCBOOT_EROM_IMG0_START = 133,
+	PROGRESS_ARCBOOT_EROM_IMG0_FINISH = 134,
 };
 
 enum security_location {
